pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -728,16 +729,50 @@ func updateBundle(tx *gorm.DB, req *datastore.UpdateBundleRequest) (*datastore.U
 		return nil, sqlError.Wrap(err)
 	}
 
-	model.Data = newModel.Data
+	if req.InputMask == nil {
+		model.Data = newModel.Data
+	} else {
+		model.Data, newBundle, err = updateBundleWithMask(model, newBundle, req.InputMask)
```

Comment:
maybe you can valiidate nil on update? to avoid this if/else?

Reply:
Oh I considered that option too. What discouraged me to do it in that way was that it would require an extra `bundleToModel()` conversion in case the input mask is nil.